### ISSUE TITLE
Behaviour of get_person_filmography

### ISSUE THREAD

**Issue by grocovid** [cinemagoer/cinemagoer#513]
#### Issue description

The behaviour of filmo=get_person_filmography seems to have changed. 
'filmography' is not a key of filmo['data'] anymore

I need to find all imdb-ids of movies with a specific person in a specific role (e.g. director)

#### Version of Cinemagoer, Python and OS

**NOTICE: please always try the latest version from the repository before submitting a bug.**

- **Python:**  3.8.1
- **Cinemagoer:**  2023.05.01
- **OS:**  Windows 6.1.7601 on x86

#### Steps to reproduce the issue

*if possible, provide a minimal code to reproduce the problem*

```python
#!/usr/bin/env python

ia=IMDb()
filmo = ia.get_person_filmography('0000080')   #Orson Welles
role='director'
if role in filmo['data']['filmography']:
   for g in filmo['data']['filmography'][role']:
     #build filmography of Orson Welles AS A DIRECTOR



#### What's the expected result?
Get the imdb-ids of movies with some person in a specific role (e.g. director)
- 

#### What's the actual result?
KeyError: 'filmography'

- 
#### Additional details
Problem started a few days ago (beginning of July 2024) - code used to work flawlessly for more than a year
-

**Comment by grocovid** [comment]
Update: I think I'm going mad.
p=ia.get_person(propernmidstring)
f=p.get("filmography")
f.get("director")

seems to do the trick... sometimes. But it seems to work a few times, than fail a few times. I can't find the logic.

**Comment by krypterro** [comment]
Now, there doesn't seem to be any filmography at all in the returned data.

**Comment by sk68877799** [comment]
@vpcreatz, seems like IMDB has changed their website completely.. No more Filmography div in the html. are you already implementing the new parser ?

**Comment by nalobinap** [comment]
@grocovid 
Hi! Have you found any solution for getting filmography data?

**Comment by meeotch** [comment]
+1 here for `KeyError: 'filmography'`.  Has imdb finally won their battle to break all the projects that make interacting with them tolerable?  :(

**Comment by grocovid** [comment]
Just checking whether there has been any progress on this. My personal movie management system relies on the possibility to list all movies from some person in a specific role (like "Orson Welles" as "director", or "Orson Welles" as "actor") It has been broken for 9 months now. I have just upgraded Cinemagoer, and checked get_person_filmography... and it doesn't seem to contain the info I need.
Can someone help? Is there any way to get the full list of movies (imdb-ids) of Orson Welles as director. Or has the IMDB eventually completely broken any attempt to retrieve meaningful data from their website?

**Comment by meeotch** [comment]
I think maybe this project has been abandoned - at least in terms of getting all the former functionality working again?  Large chunks have remained broken for a year or two, though a handful of minor fixes came in last year.

I'd love to be wrong - I, too, have built a whole media management system around this project.

**Comment by grocovid** [comment]
I see two ways around this issue (but both require some free time, something I have very little of in the foreseeable future):
* fix the scraper ourselves in Cinemagoer. It will probably require changing engine and driving a full-blown javascript-capable browser engine, which may be the reason why the original developers haven't fixed it yet.
* use the TMDB API to work around the issues, converting back and forth between the imdb-id and tmdb-id. If someone has a big hash table to convert between imdb ids and tmdb ids (for movies and persons), that would help tremendously. Otherwise, performance is going to be a drag, since a call to the api will be needed for each conversion.

In other words, it's sad that the imdb-ids have become the _de facto_ references for movies, since Amazon makes it so hard to work with them. I'm also checking on the tmdb forums whether it might be feasible to include the imdb-ids in their API output.

**Comment by filipbacic** [comment]
@grocovid Regarding imdb ids in tmdb apis, they have it already: https://developer.themoviedb.org/reference/movie-details

**Comment by grocovid** [comment]
> [@grocovid](https://github.com/grocovid) Regarding imdb ids in tmdb apis, they have it already: https://developer.themoviedb.org/reference/movie-details

Yes and no. 
- Yes at the level of a single movie. You are right on that one.
- No (unless I'm mistaken) when I ask a person filmography. So if I need - as in the use case described above - to retrieve all the movies imdb-ids tt.... for "Orson Welles" (whose imdb-id is nm0000080), I need to:
1) Convert nm0000080 to the tmbd-id (40) of Orson Welles. Easy peasy.
2) Retrieve on tmdb the filmography of Orson Welles using tmdb person id 40. One call. Easy and fast.
 BUT at this point, I get the tmdb-ids of every single movie made by Orson Welles (and his role in that movie, "cast", "director" ...)
3) So I need to loop and call the tmdb API to convert the tmdb-id of each movie to imdb-id. That'd be about 150 calls for Orson Welles !

**Comment by alberanid** [comment]
Yep, sorry for the state of the project, but besides the lack of time there are the various limitations introduced by IMDb. The "s3" data access system should still work, but contains very few data.

So, while I'm open to PR to fix things and to do some (little) work here and there, the situation is complicated.

There's the https://github.com/cinemagoer/cinemagoerng project by @uyar trying to address the IMDb limitations, and for sure it may use some help.

Regarding this version of Cinemagoer, it's possible that LLM AIs would be useful to produce/fix parsers giving them an HTML source and the parser to create or fix, but again... I had no time to test this approach.

**Comment by grocovid** [comment]
> Yep, sorry for the state of the project, but besides the lack of time there are the various limitations introduced by IMDb. The "s3" data access system should still work, but contains very few data.
> 
> So, while I'm open to PR to fix things and to do some (little) work here and there, the situation is complicated.
> 
> There's the https://github.com/cinemagoer/cinemagoerng project by [@uyar](https://github.com/uyar) trying to address the IMDb limitations, and for sure it may use some help.

Hey man, don't beat yourself up. We know how it goes with open source projects. You start with enthusiasm and - in your case - you build a terrific very useful tool with a zillion uses. And then other things happen in your life and in your head. You still fix and maintain things, but it becomes more and more tiresome. And then something happens (in this case IMDB completely rehauling their UI) and it would be extremely time-consuming to fix your tools  (and maybe it would cause regressions, dependencies that you don't want, ...). I've traveled that path (although not for something as great as imdbpy/cinemagoer). So, a million thanks for everything you gave us over the years.

Now, about the technical parts:
* cinemagoerng only provides info for the movies themselves, not persons/filmography/... . IMO, it's no more useful than cinemagoer even in its present reduced state ;
* I distrust LLMs for populating databases (they are useful for many other things of course). They will give you accurate results 99% of the time, then hallucinate some of the time, polluting your database. It's much worse than user-sources data like imdb and tmdb. Moreover, the data is not immediately in the html source anymore ( check https://www.imdb.com/fr/name/nm0000080/ and search for "the story of Samba" : it's just not there until you click on a chevron which completes the DOM). And the code is quite cryptic (obfuscated function names). So, it probably is beyond my capabilities to try and fix the imdb parser itself. And unless AIs can emulate javascript, they won't do any beteter.
* I think the best way to fix some cinemagoer functionalities is to rely more heavily on tmdb (it requires users to get an API key from there obviously). What I plan to do over the next few months or possibly couple of years is find workarounds using tmdb for my specific issues with cinemagoer. I don't know if I'm able to introduce these fixes in the code itself (I wouldn't trust myself to do it - I'm not a very good coder), but if it's ok - I'll post the code separately to retrieve things like filmographies and the like.

**Comment by alberanid** [comment]
> * I distrust LLMs for populating databases

Absolutely. The idea is not to use them to create the data, but just to fix the parsers (provided that a lot of data is simply no longer there).

LLMs do a somehow decent job with well-structured data and code, and may help a little in our case (far from being a silver bullet, for sure).

I've opened a discussion here with my 2-days testing: [use of LLM to fix parsers](https://github.com/cinemagoer/cinemagoer/discussions/530).

So far I've had some success fixing some movie-related parsers. I'll try to proceed with other parsers in the future.

**Comment by alberanid** [comment]
This issue is no longer relevant due to issue #557: ss of June 2026, Cinemagoer no longer analyzes web pages and can only rely on non-commercial datasets.
